### PR TITLE
Add ``DataFrame.map`` to ``dask.DataFrame`` API

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -3227,6 +3227,24 @@ def test_apply_warns_with_invalid_meta():
...
+    assert_eq(ddf.map(lambda x: (x, x)), df.map(lambda x: (x, x)))
+
+
+@pytest.mark.skipif(PANDAS_GT_210, reason="Available at 2.1")
+def test_dataframe_map_raises():
+    df = pd.DataFrame({"x": [1, 2, 3, 4], "y": [10, 20, 30, 40]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    with pytest.raises(
+        NotImplementedError, match="Base package doesn't support 'map'."
+    ):
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_dataframe.py:3244]: This is coming from when we try to get the `pd.DataFrame.map` docstring in the `@derived_from(pd.DataFrame)` decorator. It's nice that we get a `NotImplementedError` error, but something along the lines of (I've not tested / linted the snippet below)

```python
if not PANDAS_GT_210:
    raise NotImplementedError(
        "DataFrame.map requires pandas>=2.1.0, but pandas=X.Y.Z is installed."
    )
```

would, I think, be a bit more clear for users and devs

**Review by jrbourbeau — COMMENTED** [review]
Thanks for handling this @phofl. Overall looks good -- just left a couple of small suggestions

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -5932,6 +5932,10 @@ def apply(
     def applymap(self, func, meta="__no_default__"):
         return elemwise(M.applymap, self, func, meta=meta)
 
+    @derived_from(pd.DataFrame)
+    def map(self, func, meta="__no_default__"):
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/core.py]: Added

File: dask/dataframe/tests/test_dataframe.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py]: Yeah I had this initially :)

Changed, and removed the derived_from.

**Review by jrbourbeau — APPROVED** [review]
Thanks @phofl! Just fixed a merge conflict + included a small fixup -- this looks good to go now 👍 Will merge when CI finishes

**Pull request merged**